### DISCUSSION TITLE
Regarding FGRotor

### DISCUSSION THREAD

**Discussion by nmfarras** [JSBSim-Team/jsbsim#1163] [General]
I just check src/models/propulsion/FGRotor.cpp after reading some reference from [this](https://www.mathworks.com/help/releases/R2024b/aeroblks/rotor.html) in algorithm explanation part and [this one ](https://www.mathworks.com/help/releases/R2024b/aeroblks/computation-of-thrust-and-torque.html?searchHighlight=ctcqcomputation&s_tid=doc_srchtitle). The first link pretty much quite similar in some part of FGRotor. 

My question is does jsbsim support this? What to do in order to be able access this via xml in order to model helicoptering example.

Sorry if by chance I jump to conclusions too fast, as I still new in this and not that familiar with cpp.

Thank you as always.

**Comment by seanmcleod** [comment]
Take a look at the AH-1S helicopter model in the JSBSim repo - https://github.com/JSBSim-Team/jsbsim/tree/master/aircraft/ah1s

Here are the rotor configs used by the AH-1S.

https://github.com/JSBSim-Team/jsbsim/blob/master/engine/ah1s_rotor.xml

https://github.com/JSBSim-Team/jsbsim/blob/master/engine/ah1s_tail_rotor.xml

**Reply by nmfarras** [reply]
Ah... this makes sense now. I thought all this time it only used the engine in [this](https://github.com/JSBSim-Team/jsbsim/tree/master/aircraft/ah1s/Engines) only. 

If supposed I got information in some forum about helicopter Mil Mi 8 has on full left pedal equal to blade pitch, $`\theta_{tail}`$, is 6 degrees, on neutral 8 degree, and on full right pedal is 23 degrees, does I just need to change fcs/adj/center-sensitivity in fcs/rudder-cmd-norm-exmod based on [rotor-control](https://github.com/JSBSim-Team/jsbsim/blob/master/aircraft/ah1s/Systems/rotor_control.xml)? Does $`\theta_{tail}`$ included in rotor calculation and can be adjust with the pedal or collective mapping to blade pitch?

I kinda confuse where to edit this data. As to mapping to blade pitch, there is reference to this book appendix

 Prouty, Raymond W. Helicopter Performance, Stability, and Control Krieger Publishing Company, 1995.

**Comment by seanmcleod** [comment]
> Ah... this makes sense now. I thought all this time it only used the engine in [this](https://github.com/JSBSim-Team/jsbsim/tree/master/aircraft/ah1s/Engines) only.

JSBSim will look in both locations for engines/thrusters. Generally common components that are used by multiple aircraft types would be stored in the JSBSim `root\engine` directory and any components that are specific to just this aircraft type would be stored in `root\aircraft\aircraft-type\engines`, or sometimes users will include the engine etc. within the aircraft type's directory in order to bundle everything together.

I've submitted a pull request to move these rotor files since they are specific to this aircraft type, see - https://github.com/JSBSim-Team/jsbsim/pull/1167
